### PR TITLE
306 document the webforj war deployment method

### PR DESCRIPTION
**Pull request by MatthewHawkins**
TODO:

- [x] BLS/Licensing section
- [x] Remove hidden sidebar items on final approval of layout and update links
- [x] Update the HelloWorld code once we settle on a final version 
- [X] Add deep linking to current articles
- [X] Update maven install plugin section per [Hyyan's issue](https://github.com/webforj/webforj/issues/798)
- [X] Redirect the current getting started to the new one
- [X] For the titles, don't change them, but keep them based on what is in the sidebar
- [X] Rename hello-world to skeleton-starter-hello-world. Update this URL everywhere it's applicable
- [X] Change "latest version"  to ${webforj.version} in new getting started
- [X] Remove cards/explanation in getting started
- [X] Replace @InlineStyleSheet with @StyleSheet
- [X] Move configuration before advanced topics in the sidebar
- [X] Using BBjServices title change to Running With BBjServices, and change title of the overview article to "Installaton with BBjServices"
- [X] Rename BBj configuration to install plugin

### CODE REVIEW

**Review by hyyan — DISMISSED** [review]
@MatthewHawkins as discussed, we still the redirect from `docs/installation/docker`to the new getting started section to be handled before this get merged

**Comment by MatthewHawkins** [comment]
@hyyan Ready for your first pass review on the deployment/reload sections

File: docs/configuration/deploy-reload/jrebel.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+```bash
+mvn jetty:run
+```
+
+If done properly, JRebel will output logging information to the terminal, and changes made to your program should reflect on demand.
+
+:::info Seeing your changes
+If a change is made to a view or component that's already being displayed, JRebel won't force a reload of the page, as the server isn't restarted.
+:::
```
Comment (hyyan) [inline docs/configuration/deploy-reload/jrebel.md:23]: Before running the command document the `Add the JRebel agent to MAVEN_OPTS`
https://manuals.jrebel.com/jrebel/standalone/launch-from-command-line.html#maven-jetty-plugin-mvn-jetty-run

Comment (hyyan) [inline docs/configuration/deploy-reload/jrebel.md:23]: Remove

File: docs/configuration/deploy-reload/maven-jetty.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+The Jetty Plugin launches an embedded Jetty server that monitors your app’s files, including Java classes and resources, for changes. When it detects updates, it automatically redeploys the app, which speeds up development by eliminating manual build and deployment steps. 
+
+## Configuring the plugin
+
+Here are some essential configurations for fine-tuning the plugin’s hot deployment and server interaction settings:
+
+| Property                          | Description                                                                                                                                                                           | Default        |
+|-----------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------|
+| **`scanIntervalSeconds`**         | Configures how often the Jetty server scans for file changes in the **`pom.xml`**. The skeleton project sets this to `2` seconds. Increasing this interval can reduce CPU load but may delay changes being reflected in the app. | `2`            |
```
Comment (hyyan) [inline docs/configuration/deploy-reload/maven-jetty.md]: The property name is `scan`, and it should be included in the POM file. This configuration table is both confusing and incorrect.

Comment (hyyan) [reply docs/configuration/deploy-reload/maven-jetty.md]: The skeleton sets this to `1`

File: docs/configuration/deploy-reload/maven-jetty.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+| **`webforj.reloadOnServerError`** | When using hot redeploy, the whole WAR file is swapped. If the client sends a request while the server is restarting, an error occurs. This setting allows the client to attempt a page reload, assuming the server will be back online shortly. Only applies to development environments and only handles errors specific to hot redeployment. | `on`           |
+| **`webforj.clientHeartbeatRate`** | Sets the interval for client pings to query server availability. This keeps the client-server communication open. For development, use shorter intervals for faster error detection. In production, set this to at least 50 seconds to avoid excessive requests. | `50s`          |
+
+:::tip Development value for `webforj.clientHeartbeatRate` 
+The [`skeleton-starter-hello-world`](https://github.com/webforj/skeleton-starter-hello-world) project sets this value to `8s` for development.
```
Comment (hyyan) [inline docs/configuration/deploy-reload/maven-jetty.md]: to `1s` not `8s`

File: docs/configuration/deploy-reload/maven-jetty.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+:::
+
+
+## Usage considerations
+
+While the Jetty Plugin is highly effective for development, it has a few potential limitations:
+
+- **Memory and CPU usage**: Frequent file scanning, set by low `scanIntervalSeconds` values in the `pom.xml`, can increase resource consumption, especially on large projects. Increasing the interval may reduce load but also slows down redeployment.
```
Comment (hyyan) [inline docs/configuration/deploy-reload/maven-jetty.md]: `scan` not `scanIntervalSeconds`

File: docs/configuration/deploy-reload/maven-jetty.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+- **Limited production use**: The Jetty Plugin is designed for development, not for production environments. It lacks the performance optimization and security configurations required for production, making it better suited to local testing.
```
Comment (hyyan) [inline docs/configuration/deploy-reload/maven-jetty.md]: This no limitation. Remove

**Review by hyyan — CHANGES_REQUESTED** [review]

File: docs/configuration/deploy-reload/jrebel.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+title: JRebel
+hide_table_of_contents: true
```
Comment (hyyan) [inline docs/configuration/deploy-reload/jrebel.md:3]: Dont hide ToC for articles

File: docs/intro/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+```bash
+mvn -B archetype:generate \
+-DarchetypeGroupId=com.webforj \
+-DarchetypeArtifactId=webforj-archetype-hello-world \
+-DgroupId=org.example \
+-DarchetypeVersion=24.20 \
+-DartifactId=my-hello-world-app \
+-Dversion=1.0-SNAPSHOT
+```
```
Comment (hyyan) [inline docs/intro/getting-started.md:32]: Add a table to explain the different command params

**Pull request merged**